ui/components/app/flask/snap-settings-card/snap-settings-card.test.js

```diff
@@ -1,76 +1,40 @@
+import mockState from '../../../../../test/data/mock-state.json';
 import SnapSettingsCard from '.';
 
 describe('SnapSettingsCard', () => {
   const args = {
     name: 'Snap name',
-    description:
-      'This snap provides developers everywhere access to an entirely new data storage paradigm, even letting your programs store data autonomously.',
-    dateAdded: new Date().toDateString(),
-    version: '10.5.1234',
-    url: 'https://metamask.io',
-    status: 'stopped',
-    icon: './AST.png',
+    packageName: '@metamask/test-snap-bip44',
+    snapId: 'npm:@metamask/test-snap-bip44',
+    onClick: () => null,
   };
+  const mockStore = configureMockStore([thunk])(mockState);
+
   it('should render the SnapsSettingCard without crashing', () => {
-    const { getByText } = render(<SnapSettingsCard {...args} />);
+    const { getByText } = renderWithProvider(
+      <SnapSettingsCard {...args} />,
+      mockStore,
+    );
     expect(getByText('Snap name')).toBeDefined();
   });
 
-  it('should render the pill as installing when given a status of installing', () => {
-    args.status = 'installing';
-    const { getByText } = render(<SnapSettingsCard {...args} />);
-    expect(getByText('installing')).toBeDefined();
-  });
-
-  it('should render the pill as running when given a status of running', () => {
-    args.status = 'running';
-    const { getByText } = render(<SnapSettingsCard {...args} />);
-    expect(getByText('running')).toBeDefined();
-  });
-
-  it('should render the pill as installing when given a status of stopped', () => {
-    args.status = 'stopped';
-    const { getByText } = render(<SnapSettingsCard {...args} />);
-    expect(getByText('stopped')).toBeDefined();
-  });
-
-  it('should render the pill as crashed when given a status of crashed', () => {
-    args.status = 'crashed';
-    const { getByText } = render(<SnapSettingsCard {...args} />);
-    expect(getByText('crashed')).toBeDefined();
-  });
-
-  it('should call onToggle prop when toggle button is clicked', () => {
-    const onToggle = jest.fn();
-    args.onToggle = onToggle;
-    const { container } = render(<SnapSettingsCard {...args} />);
-    const toggleBtn = container.querySelector('.toggle-button').firstChild;
-    fireEvent.click(toggleBtn);
-    expect(onToggle).toHaveBeenCalled();
-  });
-
-  it('should call onClick prop when See Details button is clicked', () => {
-    const onClick = jest.fn();
-    args.onClick = onClick;
-    const { container } = render(<SnapSettingsCard {...args} />);
-    const seeDetailsBtn = container.querySelector(
-      '.snap-settings-card__button',
+  it('should render the icon fallback using the first letter of the name', () => {
+    const { getByText } = renderWithProvider(
+      <SnapSettingsCard {...args} icon="" />,
+      mockStore,
     );
-    fireEvent.click(seeDetailsBtn);
-    expect(onClick).toHaveBeenCalled();
+    expect(getByText('B')).toBeDefined();
```

Comment:
Could we test more clearly that this works? `getByText('B')` could resolve to any other text that includes B couldn't it?

Reply:
It looks like these React testing tools does not have extensive support for testing DOM elements in deep (like Selenium or similar). But, I've added something that should ensure that this particular case is valid with the mocked data provided at the top of the tests.